metals/src/main/scala/scala/meta/internal/metals/BuildTargetClasses.scala

```diff
@@ -1,75 +1,91 @@
 
-  def main(target: b.BuildTargetIdentifier): TrieMap[String, b.ScalaMainClass] =
-    classesOf(target).main
+  val rebuildIndex: BatchedFunction[b.BuildTargetIdentifier, Unit] =
+    BatchedFunction.fromFuture(fetchClasses)
 
-  val onCompiled: BatchedFunction[b.BuildTargetIdentifier, Unit] =
-    BatchedFunction.fromFuture(fetchMainClassesFor)
+  def classesOf(target: b.BuildTargetIdentifier): Classes = {
+    index.getOrElseUpdate(target, new Classes)
+  }
 
-  private def fetchMainClassesFor(
+  private def fetchClasses(
       targets: Seq[b.BuildTargetIdentifier]
   ): Future[Unit] = {
-    targets.foreach { target =>
-      classesOf(target).clear()
-    }
-
     buildServer() match {
       case Some(connection) =>
-        val parameters = new b.ScalaMainClassesParams(targets.asJava)
-        val task = for {
-          result <- connection.mainClasses(parameters).asScala
-          _ = cacheMainClasses(result)
-        } yield ()
+        val targetsList = targets.asJava
 
-        task
+        val updateMainClasses = connection
+          .mainClasses(new b.ScalaMainClassesParams(targetsList))
+          .thenAccept(cacheMainClasses)
+          .asScala
+
+        val updateTestSuites = connection
+          .testSuites(new b.ScalaTestClassesParams(targetsList))
+          .thenAccept(cacheTestSuites)
+          .asScala
+
+        for {
+          _ <- updateMainClasses
+          _ <- updateTestSuites
+        } yield ()
       case None =>
         Future.successful(())
     }
   }
 
-  private def classesOf(target: b.BuildTargetIdentifier): Classes =
-    index.getOrElseUpdate(target, new Classes)
-
   private def cacheMainClasses(result: b.ScalaMainClassesResult): Unit = {
-    def createObjectSymbol(className: String): String = {
-      val symbol = className.replaceAll("\\.", "/")
-      val isInsideEmptyPackage = !className.contains(".")
-      if (isInsideEmptyPackage) {
-        Symbols.Global(Symbols.EmptyPackage, Descriptor.Term(symbol))
-      } else {
-        symbol
-      }
-    }
-
     for {
       item <- result.getItems.asScala
       target = item.getTarget
       aClass <- item.getClasses.asScala
       objectSymbol = createObjectSymbol(aClass.getClassName)
-    } classesOf(target).main.put(objectSymbol, aClass)
+    } classesOf(target).mainClasses.put(objectSymbol, aClass)
   }
 
+  private def cacheTestSuites(result: b.ScalaTestClassesResult): Unit = {
+    for {
+      item <- result.getItems.asScala
+      target = item.getTarget
+      className <- item.getClasses.asScala
+      objectSymbol = createObjectSymbol(className)
+    } classesOf(target).testSuites.put(objectSymbol, className)
+  }
+
+  private def createObjectSymbol(className: String): String = {
+    val symbol = className.replaceAll("\\.", "/")
+    val isInsideEmptyPackage = !className.contains(".")
+    if (isInsideEmptyPackage) {
+      Symbols.Global(Symbols.EmptyPackage, Descriptor.Term(symbol))
+    } else {
+      symbol + "."
```

Comment:
This conversion will fail for non-ascii class names such as `HælsingSuite` or `::`. Although, I don't think these will be common, I believe the following change could make it work even in those cases

```suggestion
      import scala.reflect.NameTransformer
      symbol.split(".").foldLeft(Symbols.RootPackage) {
        case (owner, name) => Symbols.Global(owner, Descriptor.Term(NameTransformer.decode(name))
```

Either way, it would be good to have a test case to show that it does or doesn't work with non-ascii class names.

Reply:
`HælsingSuite` worked out of the box, but indeed the `::` was failing. Thanks!